tests/test_envs.py

```diff
@@ -1,27 +1,114 @@
+import re
+
 import gym
+import numpy as np
 import pytest
 
-# Import for side-effect of registering environment
+# Unused imports are for side-effect of registering environments
 import imitation.examples.airl_envs  # noqa: F401
 import imitation.examples.model_envs  # noqa: F401
 
 ENV_NAMES = [env_spec.id for env_spec in gym.envs.registration.registry.all()
              if env_spec.id.startswith('imitation/')]
 
+DETERMINISTIC_ENVS = [
+    r"imitation/CliffWorld.*-v0",
+    r"imitation/PointMaze.*-v0",
+]
 
-@pytest.mark.parametrize("env_name", ENV_NAMES)
-def test_envs(env_name):
-  """Check that our custom environments don't crash on `step`, and `reset`."""
+
+def matches_list(env_name, patterns):
+  for pattern in patterns:
+    if re.compile(pattern).match(env_name):
+      return True
+  return False
+
+
+@pytest.fixture
+def env(env_name):
   try:
     env = gym.make(env_name)
   except gym.error.DependencyNotInstalled as e:  # pragma: no cover
     if e.args[0].find('mujoco_py') != -1:
       pytest.skip("Requires `mujoco_py`, which isn't installed.")
     else:
       raise
-  env.reset()
+  return env
+
+
+@pytest.mark.parametrize("env_name", ENV_NAMES)
+def test_seed(env, env_name):
```

Comment:
Should either be renamed to reflect the fact that it only tests initial observation, or changed to take a random action before testing for observation equality (which would require seeding action space, annoyingly).

Reply:
Changed to take random sequence of actions.